src/libraries/System.Security.Cryptography.X509Certificates/tests/ExportTests.cs

```diff
@@ -294,5 +296,84 @@ bool HasEphemeralKey(X509Certificate2 c)
                 }
             }
         }
+
+        [Fact]
+        public static void TryExportCertificatePem_DestinationTooSmall()
+        {
+            using X509Certificate2 cert = new X509Certificate2(TestData.MsCertificate);
+            Assert.False(cert.TryExportCertificatePem(Span<char>.Empty, out int charsWritten));
+            Assert.Equal(0, charsWritten);
+        }
+
+        [Fact]
+        public static void TryExportCertificatePem_DestinationJustRight()
+        {
+            using X509Certificate2 cert = new X509Certificate2(TestData.MsCertificate);
+            int pemSize = PemEncoding.GetEncodedSize("CERTIFICATE".Length, cert.RawData.Length);
+            Span<char> destination = new char[pemSize];
+            Assert.True(cert.TryExportCertificatePem(destination, out int charsWritten));
+            Assert.Equal(pemSize, charsWritten);
+
+            using X509Certificate2 importedCert = X509Certificate2.CreateFromPem(destination);
+            Assert.Equal(cert.RawData, importedCert.RawData);
+        }
+
+        [Fact]
+        public static void TryExportCertificatePem_DestinationLarger()
+        {
+            using X509Certificate2 cert = new X509Certificate2(TestData.MsCertificate);
+            int pemSize = PemEncoding.GetEncodedSize("CERTIFICATE".Length, cert.RawData.Length);
+            Span<char> destination = new char[pemSize * 2];
+            Assert.True(cert.TryExportCertificatePem(destination, out int charsWritten));
+            Assert.Equal(pemSize, charsWritten);
+
+            using X509Certificate2 importedCert = X509Certificate2.CreateFromPem(destination.Slice(0, charsWritten));
+            Assert.Equal(cert.RawData, importedCert.RawData);
+        }
+
+        [Fact]
+        public static void TryExportCertificatePem_ContainsOnlyCertPem()
+        {
+            using X509Certificate2 cert = new X509Certificate2(TestData.MsCertificate);
+            int pemSize = PemEncoding.GetEncodedSize("CERTIFICATE".Length, cert.RawData.Length);
+            Span<char> destination = new char[pemSize];
+            Assert.True(cert.TryExportCertificatePem(destination, out int charsWritten));
+            Assert.Equal(pemSize, charsWritten);
+
+            int pemCount = 0;
+
+            foreach ((ReadOnlySpan<char> contents, PemFields fields) in new PemEnumerator(destination))
+            {
+                AssertExtensions.SequenceEqual("CERTIFICATE", contents[fields.Label]);
+                pemCount++;
+            }
+
+            Assert.Equal(1, pemCount);
```

Comment:
This test is what inspired me to ask the tighter export elsewhere.   Seems like we can simplify the code, and increase the amount of stuff that the test covers, if we just take an existing PEM import it, normalize the newlines to LF, and export+Assert.

```C#
[Theory]
[InlineData(true)]
[InlineData(false)]
public static void ExportCertificatePem_ContainsOnlyCertPem(bool yoda)
{
    const string Input = ...;

    string output;

    using (X509Certificate2 cert = new X509Certificate2(Encoding.ASCII.GetBytes(Input)))
    {
        if (yoda)
        {
            output = cert.ExportCertificatePem();
        }
        else
        {
            Span<char> destination = stackalloc char[Input.Length + 5];

            Assert.True(cert.TryExportCertificatePem(destination, out int charsWritten));
            Assert.Equal(Input.Length, charsWritten);

            output = destination.Slice(0, charsWritten).ToString();
        }
     }

     Assert.Equal(Input, output);
}
```

Reply:
Alternatively, we could add the "did I start at the right place?" and "did I end at the right place?" checks to what's already here.

Reply:
Hm, I see. Perhaps "ContainsOnlyCertPem" was misleading in name, the test was ensuring that (if by some miracle) a certificate's private key would not be included, or at least a form of documentation "Yes, the 'pem export' does not include CERTIFICATE nor RSA/EC PRIVATE KEY".

But I see what you are suggesting as well.

Reply:
I think we can just get rid of these tests if we tighten the others.